python/libraft/CMakeLists.txt

```diff
@@ -0,0 +1,65 @@
+# =============================================================================
+# Copyright (c) 2025, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+cmake_minimum_required(VERSION 3.26.4 FATAL_ERROR)
+
+include(../../rapids_config.cmake)
+
+project(
+  libraft-python
+  VERSION "${RAPIDS_VERSION}"
+  LANGUAGES CXX
+)
+
+option(USE_CUDA_MATH_WHEELS "Use the CUDA math wheels instead of the system libraries" OFF)
+
+# Check if raft is already available. If so, it is the user's responsibility to ensure that the
+# CMake package is also available at build time of the Python raft package.
+find_package(raft "${RAPIDS_VERSION}")
+
+if(raft_FOUND)
+  return()
+endif()
+
+unset(raft_FOUND)
+
+# --- CUDA --- #
+find_package(CUDAToolkit REQUIRED)
+set(CUDA_STATIC_RUNTIME ON)
+set(CUDA_STATIC_MATH_LIBRARIES ON)
+if(CUDAToolkit_VERSION VERSION_GREATER_EQUAL 12.0)
+  set(CUDA_STATIC_MATH_LIBRARIES OFF)
+elseif(USE_CUDA_MATH_WHEELS)
+  message(FATAL_ERROR "Cannot use CUDA math wheels with CUDA < 12.0")
+endif()
```

Comment:
Why do we need this here? Does the C++ CMakeLists.txt not handle finding CUDA appropriately? I guess it's doing some extra error handling though which is handy. Maybe this is temporary until we drop CUDA 11 and we can use CUDA math wheels for all supported CUDA versions (at which point we won't need to find the CTK in this file anymore)?

Reply:
> *Why do we need this here?*

This is here to set a value for `CUDA_STATIC_MATH_LIBRARIES`, which is closely related to the wheel-specific `USE_CUDA_MATH_WHEELS` (which controls whether or not to add RPATH entries looking for e.g. `libcublas.so` installed to `site-packages` via `nvidia-cublas-cu12`).

That isn't being newly introduced in this PR, it's just in the diff because it's being moved around. Here's where that was introduced:

* https://github.com/rapidsai/raft/pull/2376
* https://github.com/rapidsai/raft/pull/2415

> *Maybe this is temporary until we drop CUDA 11 and we can use CUDA math wheels for all supported CUDA versions (at which point we won't need to find the CTK in this file anymore)*

Yes, this.

Reply:
OK cool then that's fine.